src/math/p5.Vector.js

```diff
@@ -1400,60 +1560,91 @@ p5.Vector = class {
  * </code>
  * </div>
  */
-  setMag(n) {
-    return this.normalize().mult(n);
+  heading() {
+    const h = Math.atan2(this.y, this.x);
+    if (this.isPInst) return this._fromRadians(h);
+    return h;
   }
 
   /**
- * Calculate the angle of rotation for this vector (only 2D vectors).
- * p5.Vectors created using <a href="#/p5/createVector">createVector()</a>
- * will take the current <a href="#/p5/angleMode">angleMode()</a> into
- * consideration, and give the angle in radians or degrees accordingly.
+ * Rotates a 2D vector to a specific angle without changing its magnitude.
  *
- * @method heading
- * @return {Number} The angle of rotation
+ * If the vector was created with
+ * <a href="#/p5/createVector">createVector()</a>, `setHeading()` uses
+ * the units of the current <a href="#/p5/angleMode">angleMode()</a>.
+ *
+ * The static version of `setHeading()`, as in `p5.Vector.setHeading(v, PI)`,
+ * returns a new <a href="#/p5.Vector">p5.Vector</a> object and doesn't change
+ * the original.
+ *
+ * @method setHeading
+ * @param  {number}    angle angle of rotation.
+ * @chainable
  * @example
- * <div class = "norender">
+ * <div class="norender">
  * <code>
- * function setup() {
- *   let v1 = createVector(30, 50);
- *   print(v1.heading()); // 1.0303768265243125
+ * let v = createVector(0, 1);
+ * // Prints "1.570..." to the console.
+ * print(v.heading());
  *
- *   v1 = createVector(40, 50);
- *   print(v1.heading()); // 0.8960553845713439
+ * v.setHeading(PI);
+ * // Prints "3.141..." to the console.
+ * print(v.heading());
+ * </code>
+ * </div>
  *
- *   v1 = createVector(30, 70);
- *   print(v1.heading()); // 1.1659045405098132
- * }
+ * <div class="norender">
+ * <code>
+ * angleMode(DEGREES);
+ * let v = createVector(0, 1);
+ * // Prints "90" to the console.
+ * print(v.heading());
+ *
+ * v.setHeading(180);
+ * // Prints "180" to the console.
+ * print(v.heading());
+ * </code>
+ * </div>
+ *
+ * <div class="norender">
+ * <code>
+ * let v0 = createVector(0, 1);
+ * let v1 = p5.Vector.setHeading(v0, PI);
```

Comment:
This and below use of the static version of `setHeading()` is causing the test to fail as there is no static version of `setHeading()`

Reply:
Good catch–fixed!